src/api/app/models/project.rb

```diff
@@ -968,6 +976,44 @@ def branch_to_repositories_from(project, pkg_to_enable, opts = {})
     end
   end
 
+  def branch_remote_repositories(project)
```

Comment:
Is there a way we can add this functionality without adding another long method to the Project model? Since project.rb is already nearly 1700 lines long and looks like its being used as a [god-object](https://en.wikipedia.org/wiki/God_object). Maybe this can be extracted to a service object or some other OOP pattern? Or does this code definitely belong in the Project model?